feat: add assistant info block to system prompt

ferdinandyb opened the pull request:
Add model identity information to the system prompt, informing the assistant of its name, model ID, provider, and the current OpenCode version.

Models often misidentify themselves when asked (see #3063 where Claude Sonnet 4.5 reported itself as "Claude 3.7 Sonnet"). This adds a single sentence to the prompt to make the assistant self-aware.

Fixes #9065

ferdinandyb commented:
This would also help with this: https://bence.ferdinandy.com/2025/12/29/dont-abuse-co-authored-by-for-marking-ai-assistance/

rekram1-node commented on packages/opencode/src/session/system.ts:
```diff
@@ -31,8 +32,9 @@ export namespace SystemPrompt {
+    const provider = model ? await Provider.getProvider(model.providerID) : undefined
     return [
       [
         `Here is some useful information about the environment you are running in:`,
```
I think we should prolly put this as a first line in the system prompt rather than in environment section.

It should be something like:

You are <model> running in <OpenCode>

ferdinandyb replied:
Would that mean that each model's file would need to have a template expanded e.g like /session/prompt/anthropic.txt ? That seems to be the first line (well, anthropic_spoof for anthropic). Why are these system prompts so different actually? Are they mostly the original systemprompts tweaked a bit?

rekram1-node replied:
Instead of templating id just harcode the first sentence and remove it from all the prompts. But you could template if its easier to integrate in

Uh they are all different cause diff models need diff instructions and mostly driven from the main labs and the prompts they use with tweaks for our harness as needed

ferdinandyb replied:
@rekram1-node looking at the model specific version, it's _very_ inconsistent how the opencodeness is handled, some have instruction where to look for opencode specific stuff in detail, others just meantion it's opencode and open source.

I've pushed a new implementation injecting a new line, but maybe it would be worth to standardize the OpenCode related info there from all the models? Or those should be left as-is, because that's how the big models expect their prompts?

ferdinandyb replied:
but I think this question is still open though

ferdinandyb committed:
feat: add model self-identification to system prompt

Prepend assistant info line between header and agent prompt so models
know what they are and what version of OpenCode they're running in.

Format: "You are {name} ({provider}: {id}) running in OpenCode v{version},
the best coding agent on the planet."

ferdinandyb commented:
@rekram1-node updated the PR, also created an issue as per contributing as I think this was not needed when I first opened it?
